metaflow/plugins/retry_decorator.py

```diff
@@ -1,6 +1,22 @@
+from enum import Enum
+import os
+import signal
+import sys
+import threading
+from time import sleep
+
 from metaflow.decorators import StepDecorator
 from metaflow.exception import MetaflowException
 from metaflow.metaflow_config import MAX_ATTEMPTS
+from metaflow import current
+
+
+class RetryEvents(Enum):
```

Comment:
do we need this enum?

Reply:
its mainly for convenience when building the retry conditions on the executor implemention side, where we need to go through the supported options. Having a literal value to use instead of retyping strings reduces issues with possible typos or renames down the line.

for example this:
```
RetryEvents.PREEMPT.value in retry_conditions
```